Op bunching 2: Bunch contiguous ops for data store in a batch - data store part (#22840)
microsoft/FluidFramework: .changeset/neat-lights-worry.md

---
"@fluidframework/runtime-definitions": minor
"@fluidframework/test-runtime-utils": minor
---
---
"section": deprecation
---

> The function `process` has been deprecated on the following interface - `IFluidDataStoreChannel`. It has also been deprecated on the following mock test class - `MockFluidDataStoreRuntime`.

Reviewer: Similar to above. I think this one does want the simplification into a single sentence to be a better heading.
```suggestion
The `process` function on `IFluidDataStoreChannel` and `MockFluidDataStoreRuntime` is now deprecated
```

---
"@fluidframework/datastore-definitions": minor
"@fluidframework/runtime-definitions": minor
"@fluidframework/test-runtime-utils": minor
---
---
"section": deprecation
---

The function `process` has been deprecated on the following interface - `IFluidDataStoreChannel`. It has also been deprecated on the following mock test class - `MockFluidDataStoreRuntime`.

> A new function `processMessages` has been added in its place which will be called to process multiple messages instead of a single one. This is part of a feature called "Op bunching" where contiguous ops of a given type and to a given data store / DDS are bunched and sent together for processing.

Reviewer: For the interface case, should we explicitly call out "if you are implementing `IFluidDataStoreChannel` you must now implement `processMessages` too"? The current wording feels like "everything's handled for you" but not sure if that's accurate in the interface case. Not sure if it's actually implemented outside the repo (or supposed to, but it's not `@sealed` so it seems to be allowed).

Author: Sure, I will update it to make it explicit.

Author: Updated.

---
"section": other

> ---

Reviewer: @tylerbutler are two sections allowed in a single changeset? I don't know if the tool will handle it correctly or if this needs to be split into a second file.

Author: I wasn't sure about this one either. But I could not find any documentation or example of this.

Reviewer 2: Create two changesets if you have two different changes to describe.

Author: Updated it to two changesets
